scripts/bump_aea_version.py

```diff
@@ -18,178 +18,467 @@
-            DEFAULT_PROTOCOL_CONFIG_FILE,
-        ]
-    )
+CUR_PATH = os.path.dirname(inspect.getfile(inspect.currentframe()))  # type: ignore
+ROOT_DIR = Path(os.path.join(CUR_PATH, ".."))
+
+PLUGINS_DIR = Path("plugins")
+ALL_PLUGINS = tuple(PLUGINS_DIR.iterdir())
+
+"""
+This pattern captures a specifier set in the dependencies section
+of an AEA package configuration file, e.g.:
+
+dependencies:
+    ...
+    aea-ledger-fetchai:
+        version: >=1.0.0,<2.0.0
+"""
+YAML_DEPENDENCY_SPECIFIER_SET_PATTERN = (
+    "(?<={package_name}:\n    version: )({specifier_set})"
 )
 
-IGNORE_DIRS = [Path(".git")]
+"""
+This pattern captures a specifier set for PyPI dependencies
+in JSON format.
 
+e.g.:
+"aea-ledger-fetchai": {"version": ">=2.0.0, <3.0.0"}
+"""
+JSON_DEPENDENCY_SPECIFIER_SET_PATTERN = (
+    '(?<="{package_name}": ."version": ")({specifier_set})(?=".)'
+)
 
-def update_version_for_files(current_version: str, new_version: str) -> None:
-    """
-    Update the version.
 
-    :param current_version: the current version
-    :param new_version: the new version
-    """
-    files = [
-        Path("benchmark", "run_from_branch.sh"),
-        Path("deploy-image", "Dockerfile"),
-        Path("develop-image", "docker-env.sh"),
-        Path("docs", "quickstart.md"),
-        Path("examples", "tac_deploy", "Dockerfile"),
-        Path("scripts", "install.ps1"),
-        Path("scripts", "install.sh"),
-        Path("tests", "test_docs", "test_bash_yaml", "md_files", "bash-quickstart.md"),
-        Path("user-image", "docker-env.sh"),
-    ]
-    for filepath in files:
-        update_version_for_file(filepath, current_version, new_version)
-
-
-def update_version_for_aea(new_version: str) -> str:
-    """
-    Update version for file.
+_AEA_ALL_PATTERN = r"(?<={package_name}\[all\]==){version}"
+AEA_PATHS: PatternByPath = {
+    Path("deploy-image", "Dockerfile"): _AEA_ALL_PATTERN,
+    Path("develop-image", "docker-env.sh"): "(?<=aea-develop:){version}",
+    Path("docs", "quickstart.md"): "(?<=v){version}",
+    Path("examples", "tac_deploy", "Dockerfile"): _AEA_ALL_PATTERN,
+    Path("scripts", "install.ps1"): _AEA_ALL_PATTERN,
+    Path("scripts", "install.sh"): _AEA_ALL_PATTERN,
+    Path(
+        "tests", "test_docs", "test_bash_yaml", "md_files", "bash-quickstart.md"
+    ): "(?<=v){version}",
+    Path("user-image", "docker-env.sh"): "(?<=aea-user:){version}",
+}
 
-    :param new_version: the new version
-    :return: the current version
-    """
-    current_version = ""
-    path = Path("aea", "__version__.py")
-    with open(path, "rt") as fin:
-        for line in fin:
-            if "__version__" not in line:
-                continue
-            match = re.search('__version__ = "(.*)"', line)
-            if match is None:
-                raise ValueError("Current version is not well formatted.")
-            current_version = match.group(1)
-    if current_version == "":
-        raise ValueError("No version found!")
```

Comment:
this only checked for `__version__`; now we also check `setup.py` if the former is not present